test/python/opflow/test_cvar.py

```diff
@@ -161,6 +163,18 @@ def test_unsupported_operations(self):
             with self.assertRaises(OpflowError):
                 cvar.adjoint()
 
+    def test_cvar_on_paulisumop(self):
+        """Test a large PauliSumOp is checked for diagonality efficiently.
+
+        Regression test for Qiskit/qiskit-terra#7573.
+        """
+        op = PauliSumOp.from_list([("Z" * 30, 1)])
+        # assert global algorithm settings do not have massive calculations turned on
+        # (which is the default, but better to be sure in the test!)
+        algorithm_globals.massive = False
```

Comment:
Modifying a global variable in the test has a knock-on effect on others, because the state leaks out - even if this is the default right now, you should check the previous value, do your test in a try-finally and reset it the variable to its previous value in the `finally` branch.  If the default were ever to change, the correctness of the tests would become order-dependent, and we don't make any guarantees about that.

Reply:
I would actually use [`self.addCleanup`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addCleanup) for this instead to have the unittest framework restore the previous value after the test runs regardless of whether passes or fails. Then you don't have to worry about dealing with the error case manually.

Reply:
I don't quite get how one would use `addCleanup` to reset the value, the only thing I would have come up with is 
```python
def restore_algorithm_globals_massive(self, value):
    algorithm_globals.massive = value

def test_cvar_on_paulisumop(self):
    self.addCleanup(self.restore_algorithm_globals, algorithm_globals.massive)
    
    # previous test here
```
Is that what you had in mind?

Reply:
Yeah, that's more or less what I had in mind. I personally would probably have done something like:
```python3
self.addCleanUp(lambda x: algorithms_global.massive = x, algorithms_global.massive)
algorithm_globals.massive = False
```
but turning the lambda into a function works fine too. `addCleanup` takes a function and args to call it with and builds a LIFO queue to execute of all the addCleanups calls made during the test execution. In this case it's more straightforward though because you only have a single call and argument with a single function.

Reply:
Another possibility is using `unittest.mock.patch` as a context manager to patch the value to false for the statements you care about:
```python
with unittest.mock.patch("qiskit.utils.algorithms_global.massive", new=False):
    CVaRMeasurement(...)
    ...
```